Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +379,137 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
+		packetCount = 0;
+		PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
 
-	// try to start capture from within the callback, verify no error
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStartCapture, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+		// verify an error returns if trying capture blocking while non-blocking is running
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), 0);
+		pcpp::Logger::getInstance().enableLogs();
 
-	// try to stop capture from within the callback, verify no impact on capturing
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStopCapture, &packetCount, 10), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+		totalSleepTime = 0;
+		while (totalSleepTime <= 5)
+		{
+			pcpp::multiPlatformSleep(1);
+			totalSleepTime += 1;
+			if (packetCount > 0)
+				break;
+		}
 
-	// verify it's possible to capture non-blocking after the mess done in previous lines
-	packetCount = 0;
-	PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+		PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
 
-	// verify an error returns if trying capture blocking while non-blocking is running
-	pcpp::Logger::getInstance().suppressLogs();
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), 0);
-	pcpp::Logger::getInstance().enableLogs();
+		liveDev->stopCapture();
+		PTF_ASSERT_GREATER_THAN(packetCount, 0);
 
-	totalSleepTime = 0;
-	while (totalSleepTime <= 5)
-	{
-		pcpp::multiPlatformSleep(1);
-		totalSleepTime += 1;
-		if (packetCount > 0)
-			break;
+		liveDev->close();
+
+		// a negative test
+		pcpp::Logger::getInstance().suppressLogs();
+		PTF_ASSERT_FALSE(liveDev->startCapture(packetArrives, &packetCount));
+		pcpp::Logger::getInstance().enableLogs();
 	}
+} // TestPcapLiveDeviceBlockingMode
 
-	PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModePollTimeout)
+{
+#if !defined(_WIN32)
+	pcpp::PcapLiveDevice* liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
+	auto interfaceName = liveDev->getName();
 
-	liveDev->stopCapture();
-	PTF_ASSERT_GREATER_THAN(packetCount, 0);
+	// drop all packets on the interface
+	auto iptablesAddInputDrop = "sudo iptables -A INPUT -i " + interfaceName + " -j DROP";
+	auto iptablesAddOutputDrop = "sudo iptables -A OUTPUT -o " + interfaceName + " -j DROP";
+	if(std::system(iptablesAddInputDrop.c_str()) < 0)
+		throw std::runtime_error("failed to run: " + iptablesAddInputDrop);
+	if(std::system(iptablesAddOutputDrop.c_str()) < 0)
+		throw std::runtime_error("failed to run: " + iptablesAddOutputDrop);
```

Comment:
This shouldn't throw an exception. Instead - it should fail:
```cpp
PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddInputDrop.c_str()), 0);
PTF_ASSERT_GREATER_OR_EQUAL_THAN(std::system(iptablesAddOutputDrop.c_str()), 0);
```

Reply:
do you have a better way for the following code? it's in `TestUtils.h`, but I think we shouldn't include `PcppTestFramework.h` into it.
```cpp
~SystemCommandTeardown() noexcept(false)
	{
		if (!m_CancelTeardown)
		{
			if(std::system(m_command.c_str()) < 0)
			{
				// cppcheck-suppress exceptThrowInDestructor
				throw std::runtime_error("failed to run: " + m_command);
			}
		}
	}
```

Reply:
We currently don't have a graceful way to exit teardown functions. I guess that throwing an exception is ok...

Reply:
sure.